kmm/kmm-post-data/src/commonMain/kotlin/com/gchristov/newsfeed/kmmpostdata/usecase/DecoratePostUseCase.kt

```diff
@@ -0,0 +1,74 @@
+package com.gchristov.newsfeed.kmmpostdata.usecase
+
+import com.gchristov.newsfeed.kmmpostdata.Post
+import com.gchristov.newsfeed.kmmpostdata.PostRepository
+import com.gchristov.newsfeed.kmmpostdata.model.DecoratedPost
+import com.gchristov.newsfeed.kmmpostdata.util.ReadingTimeCalculator
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+import kotlinx.datetime.Instant
+
+class DecoratePostUseCase(
+    private val postRepository: PostRepository,
+    private val dispatcher: CoroutineDispatcher
+) {
+
+//    suspend fun decoratedPost(
+//        postId: String
+//    ): DecoratedPost {
+//
+//        return postRepository.run {
+//            cachedPost(postId)?.let { post ->
+//                clearCache(postId)
+//                decoratePost(post)
+//            } ?: queryNewPost(postId)
+//        }
+//    }
+
+    suspend fun decoratedPost(
+        postId: String
+    ): DecoratedPost {
+
+        return postRepository.run {
+//            cachedPost(postId)?.let { post ->
+//                clearCache(postId)
+//                decoratePost(post)
+//            } ?:
+            queryNewPost(postId)
+        }
+    }
+
+    suspend fun cachedPost(postId: String): DecoratedPost? =
+        postRepository.cachedPost(postId)?.let { post ->
+            decoratePost(post)
+        }
+
+    suspend fun clearCache(postId: String) = postRepository.clearCache(postId)
```

Comment:
These feel like it should be delegated to the `PostRepository` instead of here.

Since this is already a use case, we can can have a cache callback in `decoratedPost` method which will call the post repository and notify the caller of any available cache. This is the current approach for the feed at the moment (see my comment above)

Reply:
This one should've now been addressed

Reply:
This should've been done